feat: rework trigger workflow docs (#764)
knocklabs/docs: content/send-notifications/triggering-workflows/overview.mdx

---
title: Triggering workflows
description: Learn more about how to trigger cross-channel notification workflows in Knock.
tags: ["trigger", "notify", "data", "actor"]
section: Send notifications
---

> Workflows that you design within Knock are triggered, which executes a workflow run for one or more recipients that are specified. Workflows can be triggered in a number of ways:

Reviewer: ```suggestion
"Triggering" a workflow that you design within Knock executes it to run for one or more recipients you specify. Workflows can be triggered in a number of ways:
```

- By a [direct API call](/send-notifications/triggering-workflows/api) to the `trigger` endpoint
- By a [source event](/send-notifications/triggering-workflows/source-events)
- On a [recurring, or one-off schedule](/send-notifications/triggering-workflows/schedules)
- When a user becomes [a member of an audience](/send-notifications/triggering-workflows/audiences)

> It's important to realize that triggering a workflow in Knock may result in no messages being sent to your recipients. This is because your recipients may have indicated through their `preferences` that they don't wish to be notified by workflows of that type. The good news is that Knock handles all [preference-based opt-outs](/preferences/overview) for you automatically.

Reviewer: ```suggestion
**Important:** triggering a workflow in Knock may result in no messages being sent to your recipients. This is because your recipients may have indicated through their `preferences` that they don't wish to be notified by workflows of that type. Knock handles all [preference-based opt-outs](/preferences/overview) for you automatically.
```

## Conditionally executing a workflow trigger

> A trigger step on a workflow can have one or more [step conditions](/designing-workflows/step-conditions), which determine if the workflow will execute. When the conditions are evaluated as false, then the workflow will terminate and no other steps will be executed.

Reviewer: ```suggestion
A trigger step on a workflow can have one or more [step conditions](/designing-workflows/step-conditions) which determine if the workflow will execute. When the conditions are evaluated as false, then the workflow will terminate and no other steps will be executed.
```

## Controlling workflow trigger frequency

> In some situations, you may want to limit the number of times that a recipient can run through a workflow. For example, for an account signup workflow, you may want to ensure that a recipient can only ever run through the workflow once. That's where workflow trigger frequency comes in.

Reviewer: ```suggestion
In some situations, you may want to limit the number of times that a workflow will run for a recipient. For example, for an account signup workflow, you may want to ensure that a workflow will only be run once for a recipient. That's where workflow trigger frequency comes in.
```
